src/ipoplib.py

```diff
@@ -626,21 +747,46 @@ def parse_config():
     if not ("xmpp_username" in CONFIG and "xmpp_host" in CONFIG):
         raise ValueError("At least 'xmpp_username' and 'xmpp_host' must be "
                          "specified in config file or string")
-
+    # Use custom keyring incase of non-interactive environment
+    # (uses a fixed password)
+    if isinstance(keyring.get_keyring(),keyring.backends.file.EncryptedKeyring):
```

Comment:
keyring module, according to various parameters of the system, will chose the best keyring. The option that is chosen last is plaintext keyring. I had that condition to avoid keyring prompting for password in non-interactive systems. Incase of plaintext keyring, it will not prompt for password.